git-examples.md

```diff
@@ -0,0 +1,63 @@
+# GIT Examples
+## Adding a feature to PX4
+
+Adding a feature to PX4 follows a defined workflow. In order to share your contributions on PX4, you can follow this example.
+
+* [Sign up](https://github.com/join) for github if you haven't already
+* Fork the Firmware (see [here](https://help.github.com/articles/fork-a-repo/#fork-an-example-repository))
+* Clone your fork to your local computer<br>
+```sh
+git clone https://github.com/<your git name>/Firmware.git
+```
+* Go into the new directory, initialize and update the submodules, and add the original upstream Firmware<br>
+```sh
+cd Firmware
+git submodule update --init --recursive
+git remote add upstream https://github.com/PX4/Firmware.git
+```
+* Make the changes that you want to add
+* Create a new branch with a meaningful name that represents your feature<br>
+```sh
+git checkout -b <your branch name>
+```
+you can use the command ```git branch``` to make sure you're on the right branch
+* Add your changes that you want to be part of the commit<br>
+```sh
+git add .
```

Comment:
I would not suggest that. It usually does not end up in a good commit (as you probably end up having unrelated changes in one commit). Using a gui (like git gui [0]), or `git add -p` [1] would IMO be prefered. This way you review your own changes one more time.

[0] https://git-scm.com/book/en/v2/Git-in-Other-Environments-Graphical-Interfaces
[1] http://nuclearsquid.com/writings/git-add/